solarkraft/src/instrument.ts

```diff
@@ -1,101 +1,66 @@
+import { ContractCallEntry } from './fetcher/storage.js'
+
 /**
  * @license
  * [Apache-2.0](https://github.com/freespek/solarkraft/blob/main/LICENSE)
  */
-type Value = { type: string; value: any }
-type Binding = { name: string; type: string; value: any }
-type State = Binding[]
-type Transaction = {
-    functionName: string
-    functionArgs: Value[]
-    env: { timestamp: number }
-    error: string
-}
-
-// Return true iff `o` is a Javascript object.
-function isObject(o: any) {
-    return typeof o === 'object' && !Array.isArray(o) && o !== null
-}
-
-// Decode an ITF value `value` into a Javascript value.
-// TODO(#46): support composite types (sequence, tuple, record, set, map)
-function decodeITFValue(value: any) {
-    if (
-        isObject(value) &&
-        Object.prototype.hasOwnProperty.call(value, '#bigint')
-    ) {
-        return parseInt(value['#bigint'])
-    }
-    return value
-}
-
-/**
- * Return a `State` from an ITF JSON object.
- * @param itf ITF JSON object, see https://apalache.informal.systems/docs/adr/015adr-trace.html
- * @returns `State`
- */
-export function stateFromItf(itf: any): State {
-    // const vars = itf['vars'] // e.g., [ "is_initialized", "last_error" ]
-    const varTypes = itf['#meta']['varTypes'] // e.g., { "is_initialized": "Bool", "last_error": "Str" }
-    const initState = itf['states'][0] // e.g., [ { "is_initialized": false, "last_error": "" }, state1, ... ]
-    delete initState['#meta']
-
-    const state = Object.entries(initState)
-        .filter(([name]) => name != '#meta')
-        .map(([name, value]) => ({
-            name: name,
-            value: decodeITFValue(value),
-            type: 'Tla' + varTypes[name],
-        }))
-    return state
-}
 
 /**
  * Return an instrumented monitor specification.
  * @param monitor A TLA+ monitor, as returned from `apalache typecheck --output=temp.json`
- * @param state The blockchain state before executing the transaction / the monitor.
- * @param tx The transaction being applied to `state`.
+ * @param contractCall The contract call, fetched from the ledger.
  * @returns The instrumented TLA+ monitor.
  */
 export function instrumentMonitor(
     monitor: any,
-    state: State,
-    tx: Transaction
+    contractCall: ContractCallEntry
 ): any {
     // Only instrument state variables that are delcared in the monitor spec
     // (otherwise we provoke type errors in Apalache Snowcat, via undeclared variables)
     const declaredMonitorVariables = monitor.modules[0].declarations
         .filter(({ kind }) => kind == 'TlaVarDecl')
         .map(({ name }) => name)
-    state = state.filter(({ name }) => declaredMonitorVariables.includes(name))
+    const fieldsToInstrument = contractCall.oldFields.filter((value, key) =>
```

Comment:
Is it `(value, key)` or `(key, value)`? Looking at the file in `.stor` here, it seems to be the opposite.